sdk/spring/scripts/update_external_dependencies_by_spring.py

```diff
@@ -102,9 +108,12 @@ def update_dependency_dict(dependency_dict, root_pom_id):
             artifact_type = dependency_element.find('./maven:type', MAVEN_NAME_SPACE)
             if artifact_type is not None and artifact_type.text.strip() == 'pom':
                 new_pom = Pom(group_id, artifact_id, version, pom.depth + 1)
-                q.put(new_pom)
-                pom_count = pom_count + 1
-                log.debug('Added new pom. depth = {}, url = {}.'.format(new_pom.to_url(), new_pom.depth))
+                new_pom_string = '{}:{};{}'.format(group_id, artifact_id, version)
```

Comment:
Update `__str__(self)` in `pom.py`:

```
    def __str__(self):
        return '{}:{};{}'.format(self.group_id, self.artifact_id, self.version)
```
i.e. Change `{}:{}:{}` to `{}:{};{}`, second `:` change to `;`.

Then we can use `str(new_pom)` instead of define `new_pom_string`

Reply:
And update `output_version_dict_to_file(...)`, use `'{};{}\n'` to replace `'{}:{}\n'`.

Reply:
1. Rename `ROOT_POM_IDS` to `ROOT_POMS`.
2. Delete `SKIP_IDS`